main.go

```diff
@@ -73,6 +73,12 @@ func (c Config) walkSequenceNode(in *yaml.RNode) error {
 
 func (c Config) walkMapNode(in *yaml.MapNode) error {
 	key, err := in.Key.GetString(".")
+
+	_, err = c.Filter(in.Key)
+	if err != nil {
+		return err
+	}
+
```

Comment:
Actually, retrieving the key name should probably come after substituting it otherwise the key passed in "AppendToFieldPath" below will be the wrong value.
Although it didn't appear to affect the outcome when I ran it 🤷‍♂️
```suggestion
	_, err = c.Filter(in.Key)
	if err != nil {
		return err
	}

	key, err := in.Key.GetString(".")

```

Reply:
It's used by the fieldType package so could result in not edge cases around handling strings that look like bools and numbers if not done right